src/core/runtime/runtime.h

```diff
@@ -52,9 +52,10 @@ struct Core {
    * @brief Performs a registration callback. Libraries must perform
    * registration of tasks and other components through this function.
    *
-   * @param callback Registration callback to perform
+   * @tparam CALLBACK Registration callback to perform
    */
-  static void perform_registration(RegistrationCallback callback);
```

Comment:
Obviously we're going to update cunumeric ASAP, but as a matter of CI and deprecation best practice, I would probably keep the old callback and maybe mark it deprecated.  In this case, you can actually give the new and old callbacks the same name since C++ resolution allows functions with the same name where one is templated and the other is not.

Reply:
we've been using the branch name trick to match up PRs (maybe you didn't notice, but there's a companion PR for cuNumeric here: https://github.com/nv-legate/cunumeric/pull/827). and I think we continue doing something similar with the help from CI, instead of phasing out obsolete APIs like you suggested. Legate is still being actively developed, so I agree that we need a proper deprecation protocol, but I feel that'd just complicate the process for the time being.

Reply:
Sounds good.